File: typescript/45-Jump-Game-II.ts
function jump(nums: number[]): number {
    let left = 0;
    let right = 0;
    let res = 0;

    while (right < nums.length - 1) {
        let maxJump = 0;
        for (let i = left; i <= right; i++) {
            maxJump = Math.max(maxJump, i + nums[i]);
        }
        left = right + 1;
        right = maxJump;
        res += 1;
    }
    return res;
}

File: c/45-Jump-Game-II.c
int jump(int* nums, int numsSize){
    int left = 0;
    int right = 0;
    int res = 0;
    
    while (right < numsSize - 1) {
        int maxJump = 0;
        for (int i = left; i <= right; i++) {
            maxJump = max(maxJump, i + nums[i]);
        }
        left = right + 1;
        right = maxJump;
        res += 1;
    }
    return res;
}

// C doesn't have a built-in max function
int max(int a, int b) {
    return (a > b) ? a : b;
}